File: LMIS/LMIS.Web/Modules/BookManage/Authors/AuthorsForm.cs
using LMIS.Modules.Common.Enums;
using Serenity.ComponentModel;
using System;

namespace LMIS.BookManage.Forms;

[FormScript("BookManage.Authors")]
[BasedOnRow(typeof(AuthorsRow), CheckNames = true)]
public class AuthorsForm
{
    public string AuthorName { get; set; }
    public SexEnum Sex { get; set; }
    public string Email { get; set; }
    public string AuthorImage { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/AuthorsForm.ts
import { StringEditor, EnumEditor, EmailAddressEditor, ImageUploadEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { SexEnum } from "../Modules/Common.Enums.SexEnum";

export interface AuthorsForm {
    AuthorName: StringEditor;
    Sex: EnumEditor;
    Email: EmailAddressEditor;
    AuthorImage: ImageUploadEditor;
}

export class AuthorsForm extends PrefixedContext {
    static readonly formKey = 'BookManage.Authors';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AuthorsForm.init)  {
            AuthorsForm.init = true;

            var w0 = StringEditor;
            var w1 = EnumEditor;
            var w2 = EmailAddressEditor;
            var w3 = ImageUploadEditor;

            initFormType(AuthorsForm, [
                'AuthorName', w0,
                'Sex', w1,
                'Email', w2,
                'AuthorImage', w3
            ]);
        }
    }
}

[SexEnum]; // referenced types